index.js

```diff
@@ -76,3 +89,78 @@ export default function makeAsynchronous(function_) {
 		}
 	});
 }
+
+const makeIteratorContent = function_ =>
+	`
+	let nothing = Symbol('nothing');
+	let iterator = nothing;
+
+	globalThis.onmessage = async ({data: arguments_}) => {
+		try {
+			if (iterator === nothing) {
+				iterator = await (${function_.toString()})(...arguments_);
+			}
+
+			const output = await iterator.next();
+			globalThis.postMessage({output});
+		} catch (error) {
+			globalThis.postMessage({error});
+		}
+	};
+	`;
+
+export function makeAsynchronousIterator(function_) {
+	return (...arguments_) => ({
+		async * [Symbol.asyncIterator]() {
+			let worker;
+			let cleanup;
+
+			const failure = error => {
+				cleanup();
+				throw error;
```

Comment:
The throw here will bubble up to the event handler and not the body of the iterator. For example, line 137.